### ISSUE TITLE
follow up on Collection Activity Tab

### ISSUE THREAD

**Issue by daiagi** [kodadot/nft-gallery#5388]
to sum it up, left overs for next PR:

- [ ] display sensible data in the chart #5362 
- [ ] holder/flipper info on RMRK2
- [ ] use ```formatDistanceFromNow``` instead of ```timeAgo``` https://github.com/kodadot/nft-gallery/pull/5345#discussion_r1148504226

* #5345

**Comment by exezbcz** [comment]
scroll position reset when switching tabs (items/activity)

**Comment by daiagi** [comment]
by @roiLeo : https://github.com/kodadot/nft-gallery/pull/5345#pullrequestreview-1358436762
- I would remove `import { set } from 'vue'`
- make `has-text-k-blue` & `has-text-k-grey` global [variables](https://bulma.io/documentation/customize/variables/) class
- query doesn't have limit on events
- put composable (`useReplaceUrl`) in composable folder 
- relative path
- duplicate css code & some bad selector (`.background-color`??)

**Comment by yangwao** [comment]
Plus codeclimate review 
- https://github.com/kodadot/nft-gallery/pull/5345#issuecomment-1484620079
- https://codeclimate.com/github/kodadot/nft-gallery/pull/5345

**Comment by daiagi** [comment]
@roiLeo
About the point you've raised that there is no limit on events
I think it is necessary to get all events in order to have correct flippers info
I guess a way to reduce data is to get all mints and all send/ buy events
And paginate the rest as user scrolls down the list

But I don't that willy make much difference in query size


Wdyt?

**Comment by roiLeo** [comment]
> @roiLeo About the point you've raised that there is no limit on events I think it is necessary to get all events in order to have correct flippers info I guess a way to reduce data is to get all buy events And paginate the rest as user scrolls down the list

Yes! Limit 20-50 and paginate the rest.

> But I don't that will make much difference in query size
> 
> Wdyt?

Let's imagine I make 100k+ "LIST" event, client side won't be able to handle all this data.

**Comment by daiagi** [comment]
OK. Just to clarify...
You do agree that all buy events will still be needed to be fetched immediately, right?

**Comment by roiLeo** [comment]
> OK. Just to clarify... You do agree that all buy events will still be needed to be fetched immediately, right?

for me not necessarily. Is it for the graph? this would be the events over a given period of time.

**Comment by daiagi** [comment]
The graph too I guess 
But I was thinking about flippers tab, for that I need to know about every time an nft switch hands, so that means all buy events

By the way, on previous collection activity minters who sold their nft are also dubbed flippers, I disagree with that.
I think only someone who bought and sold is a flipper. 
 For now I kept this as it was before

 Though

**Comment by daiagi** [comment]
> scroll position reset when switching tabs (items/activity) 
> 


Fixed by #5416

**Comment by daiagi** [comment]
@vikiival I think creating a userEntity is the way forward
A userEntity will have all the data required for holders flippers tab and will take the load of calculating it off the front end.
Does that make sense?

Thank you

**Comment by roiLeo** [comment]
> @vikiival @roiLeo I think creating a userEntity is the way forward A userEntity will have all the data required for holders flippers tab and will take the load of calculating it off the front end. Does that make sense?

Yes! kodadot/rubick#52 & kodadot/rubick#56 (maybe kodadot/rubick#157 for a start)
Feel free to try to write new schema, I can help you if you need to setup indexer project.

**Comment by daiagi** [comment]
it will take some time as in in vacation until 9th of April

If you are interested in getting it going feel free to, otherwise I will try my strength in it when I'm back

**Comment by vikiival** [comment]
> 9th of April

Will take that time to design new Schema 2.0 😅

**Comment by yangwao** [comment]
> limit events in query and paginate

Only this is left in this issue?

**Comment by roiLeo** [comment]
> > limit number of events in query, paginate events table
> 
> Only this is left in this issue?

Yes imo same stuff as #5540

**Comment by yangwao** [comment]
this is related I guess
- https://github.com/kodadot/nft-gallery/issues/5574

Good for resolver?

**Comment by yangwao** [comment]
hey @daiagi can we look on this as we want to launch rmrk2 bug hunting 

- #5495 
- https://github.com/kodadot/nft-gallery/issues/5536

**Comment by daiagi** [comment]
hey @yangwao 
it is waiting on this:
* https://github.com/kodadot/loligo/issues/5

once we can fetch holder/flippers info as part of new ```userEntity```, the need to fetch all events on FE in order to calculate them will be eliminated. then we can paginate the events table in activity tab

**Comment by daiagi** [comment]
events table is already lazy loaded 
I've lazy loaded holders/flippers tabs in #5608  and seeing performance improvement

**Comment by JustLuuuu** [comment]
@roiLeo can we close this?

**Comment by roiLeo** [comment]
> @roiLeo can we close this? 

Im not aware of current implementation, it was needed to limit number of events in query so user won't face potential bug where there is more than 1k events.

Feel free to close it.
If a bug occur on a big collection it could be related to that one

**Comment by vikiival** [comment]
Still an issue but we would need resolver
